pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -795,6 +818,42 @@ func TestPlugin(t *testing.T) {
 			},
 			disableDRA: true,
 		},
+		"claim-with-request-with-unknown-device-class": {
+			pod:    podWithClaimName,
+			claims: []*resourceapi.ResourceClaim{updateDeviceClassName(claim, "does-not-exist")},
+			want: want{
+				prefilter: result{
+					status: framework.NewStatus(framework.Unschedulable, `request req-1: device class does-not-exist does not exist`),
+				},
+				postfilter: result{
+					status: framework.NewStatus(framework.Unschedulable, `no new claims to deallocate`),
+				},
+			},
+		},
+		"claim-with-prioritized-list": {
+			pod:     podWithClaimName,
+			claims:  []*resourceapi.ResourceClaim{claimWithPrioritzedList},
+			classes: []*resourceapi.DeviceClass{deviceClass},
+			want: want{
+				filter: perNodeResult{
+					workerNode.Name: {
+						status: framework.NewStatus(framework.UnschedulableAndUnresolvable, `claim default/my-pod-my-resource, request req-1: has subrequests, but the feature is disabled`),
+					},
+				},
+			},
+		},
+		"claim-with-prioritized-list-unknown-device-class": {
+			pod:    podWithClaimName,
+			claims: []*resourceapi.ResourceClaim{updateDeviceClassName(claimWithPrioritzedList, "does-not-exist")},
+			want: want{
+				prefilter: result{
+					status: framework.NewStatus(framework.Unschedulable, `request req-1/subreq-0: device class does-not-exist does not exist`),
+				},
+				postfilter: result{
+					status: framework.NewStatus(framework.Unschedulable, `no new claims to deallocate`),
+				},
+			},
+		},
```

Comment:
One positive test case where allocation with `firstAvailable` succeeds would be good here. We don't need to cover all possible scenarios (that's over in allocator_test.go), just one.

Reply:
Added